components/brave_news/browser/feed_fetcher.h

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_NEWS_BROWSER_FEED_FETCHER_H_
+#define BRAVE_COMPONENTS_BRAVE_NEWS_BROWSER_FEED_FETCHER_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/containers/flat_map.h"
+#include "base/functional/callback_forward.h"
+#include "base/memory/weak_ptr.h"
+#include "brave/components/api_request_helper/api_request_helper.h"
+#include "brave/components/brave_news/browser/channels_controller.h"
+#include "brave/components/brave_news/browser/direct_feed_fetcher.h"
+#include "brave/components/brave_news/browser/publishers_controller.h"
+#include "brave/components/brave_news/common/brave_news.mojom.h"
+
+namespace brave_news {
+
+using FeedItems = std::vector<mojom::FeedItemPtr>;
+using ETags = base::flat_map<std::string, std::string>;
+using FetchFeedCallback = base::OnceCallback<void(FeedItems items, ETags tags)>;
+using UpdateAvailableCallback = base::OnceCallback<void(bool)>;
+
+class FeedFetcher {
+ public:
+  explicit FeedFetcher(
```

Comment:
only need `explicit` for ctor with single parameter.

```suggestion
  FeedFetcher(
```

Reply:
Nice catch!